rpcs3/Emu/Cell/Modules/cellAtracXdec.cpp

```diff
@@ -0,0 +1,891 @@
+				{
+					break;
+				}
+			}
+
+			atracx_param = cmd.atracx_param;
+			break;
+		}
+		case AtracXdecCmdType::end_seq:
+		{
+			skip_getting_command = true;
+
+			// Block savestate creation during callbacks
+			std::unique_lock savestate_lock{g_fxo->get<hle_locks_t>(), std::try_to_lock};
+
+			if (!savestate_lock.owns_lock())
+			{
+				ppu.state += cpu_flag::again;
+				return;
+			}
+
+			skip_getting_command = false;
+
+			// Doesn't do anything else
+			notify_seq_done.cbFunc(ppu, notify_seq_done.cbArg);
+			break;
+		}
+		case AtracXdecCmdType::decode_au:
+		{
+			skip_getting_command = true;
+
+			ensure(!!cmd.au_start_addr); // Not checked on LLE
+
+			cellAtracXdec.trace("Waiting for output to be consumed...");
+
+			lv2_obj::sleep(ppu);
+			std::unique_lock output_mutex_lock{output_mutex};
+
+			while (output_locked && !ppu.is_stopped())
+			{
+				lv2_obj::sleep(ppu);
+				output_consumed.wait(output_mutex, 20000);
+			}
+
+			if (ppu.is_stopped())
+			{
+				ppu.state += cpu_flag::again;
+				return;
+			}
+
+			if (!run_thread)
+			{
+				return;
+			}
+
+			cellAtracXdec.trace("Output consumed");
+
+			u32 error = CELL_OK;
+
+			// Only the first valid ATS header after starting a sequence is parsed. It is ignored on all subsequent access units
+			if (first_decode && atracx_param.au_includes_ats_hdr_flg == CELL_ADEC_ATRACX_ATS_HDR_INC)
+			{
+				// Block savestate creation during callbacks
+				std::unique_lock savestate_lock{g_fxo->get<hle_locks_t>(), std::try_to_lock};
+
+				if (!savestate_lock.owns_lock())
+				{
+					ppu.state += cpu_flag::again;
+					return;
+				}
+
+				if (error = decoder.parse_ats_header(cmd.au_start_addr); error != CELL_OK)
+				{
+					notify_error.cbFunc(ppu, error, notify_error.cbArg);
+				}
+				else if (decoder.init_decode(atracx_param.bw_pcm, atracx_param.nch_out) != CELL_OK)
+				{
+					notify_error.cbFunc(ppu, CELL_ADEC_ERROR_ATX_FATAL, notify_error.cbArg);
+				}
+			}
+
+			// LLE does not initialize the output address if parsing the ATS header fails
+			vm::ptr<void> output = vm::null;
+			u32 decoded_samples_num = 0;
+
+			if (error != CELL_ADEC_ERROR_ATX_ATSHDR)
+			{
+				// The LLE SPU thread would crash if you attempt to decode without a valid configuration
+				ensure(decoder.config_is_set, "Attempted to decode with invalid configuration");
+
+				output.set(work_mem.addr() + atracXdecGetSpursMemSize(decoder.nch_in));
+
+				const auto au_start_addr = atracx_param.au_includes_ats_hdr_flg == CELL_ADEC_ATRACX_ATS_HDR_INC ? cmd.au_start_addr.get_ptr() + sizeof(AtracXdecAtsHeader) : cmd.au_start_addr.get_ptr();
+				std::memcpy(work_mem.get_ptr(), au_start_addr, decoder.nbytes);
+
+				if (int err = avcodec_send_packet(decoder.ctx, decoder.packet); err)
+				{
+					// These errors should never occur
+					if (err == AVERROR(EAGAIN) || err == -static_cast<s32>("EOF "_u32) /* AVERROR_EOF */ || err == AVERROR(EINVAL) || err == AVERROR(ENOMEM))
```

Comment:
Just do
constexpr int averror_eof = AVERROR_EOF; // workaround for old-style-cast error
somewhere and use that one.